Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -801,7 +801,8 @@ function Set-AvsVMStoragePolicy {
         $VMName
     )
     Write-Host "Getting Storage Policy $StoragePolicyName"
-    $StoragePolicy =  Get-SpbmStoragePolicy -Namespace "VSAN" -ErrorAction Stop | Where-Object {$_.Name -eq $StoragePolicyName}
+    $VSANStoragePolicies = Get-SpbmStoragePolicy -Namespace "VSAN" -ErrorAction Stop
+    $StoragePolicy =  $VSANStoragePolicies | Where-Object {$_.Name -eq $StoragePolicyName}
     if ($null -eq $StoragePolicy) {
         Write-Error "Could not find Storage Policy with the name $StoragePolicyName. It either does not exist or is not available." -ErrorAction Continue
         Write-Error "Available storage policies: $(Get-SpbmStoragePolicy -Namespace "VSAN")" -ErrorAction Stop
```

Comment:
May be better to read it once and/or pull it into a helper function to avoid querying with the same string arg?